torch/distributed/_composable/replicate.py

```diff
@@ -0,0 +1,65 @@
+import torch
+import torch.nn as nn
+
+from typing import List, Tuple
+
+
+class DistributedState:
+    ...
+
+
+class ReplicateState(DistributedState):
+    def __init__(self) -> None:
+        self.modules: List[nn.Module] = []
+        self.parameters: List[nn.Parameter] = []
```

Comment:
The `replicate()` API can be called multiple times for different modules in a model. This `parameter` field is to keep a reference of all parameters of these modules. So that they can be managed together, for example bucketizing in DDP's all-reduce.